src/wrapper.rs

```diff
@@ -123,26 +139,25 @@ macro_rules! glib_wrapper {
 
     (
         $(#[$attr:meta])*
-        pub struct $name:ident(Shared<$ffi_name:path>);
+        pub struct $name:ident(Object<$ffi_name:path>);
 
         match fn {
-            ref => |$ref_arg:ident| $ref_expr:expr,
-            unref => |$unref_arg:ident| $unref_expr:expr,
+            get_type => || $get_type_expr:expr,
         }
     ) => {
-        glib_shared_wrapper!([$($attr)*] $name, $ffi_name, @ref $ref_arg $ref_expr,
-            @unref $unref_arg $unref_expr);
+        glib_object_wrapper!([$($attr)*] $name, $ffi_name, @get_type $get_type_expr, []);
     };
 
     (
         $(#[$attr:meta])*
-        pub struct $name:ident(Object<$ffi_name:path>);
+        pub struct $name:ident(Object<$ffi_name:path>): [$($implements:tt)+];
```

Comment:
`[]` lets me capture its insides as a set of `tt`, that can be processed later. It's difficult if not impossible to work with a variable number of tokens somewhere other than at the right side of a token tree. `[]` provides a separate tree, without `[]` I'd get ambiguity with `;`.

So, without `[]` I'd have to make each entry use the extended syntax. Maybe that's not a big downside. Also note that `path` must be followed by either of `=>` `,` `=` `|` `;` `:` `>` `[` `{` `as` `where`.